### PR TITLE
Use `@antfu/ni` to resolve build command

### PR DESCRIPTION
**Pull request by ghengeveld**
Rather than rolling our own package manager script runner logic, use `@antfu/ni` to handle it for us. This adds support for pnpm and Bun, since `ni` supports those too.

Note: an enhancement would be to upgrade Execa to the same version `ni` uses, but I did not go down that road yet because it upgraded to ESM and therefore isn't compatible with our current Jest config. I'd love to fix that but it would be a larger undertaking.

### CODE REVIEW

**Review by thafryer — COMMENTED** [review]
Largely looks good to me. Few notes to help improve understanding. 

Also, looking at the ticket that this work is assigned to, shouldn't we also add the pnpm lockfile as file that TurboSnap bails on?

File: node-src/tasks/build.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import { parseNr, getCliCommand, parseNa } from '@antfu/ni';
```
[the remark this answers is not in the document]
Comment (ghengeveld) [reply node-src/tasks/build.ts:1]: I extracted the logic to some utility functions with better names.

**Comment by ghengeveld** [comment]
This has been released as [7.0.0-canary.0](https://www.npmjs.com/package/chromatic/v/7.0.0-canary.0) available under the `canary` tag (`chromatic@canary`) on npm and the `chromaui/action-canary` GitHub action.

**Review by ndelangen — APPROVED** [review]

**Comment by glancel** [comment]
Hi! How can we help to test if pnpm works? If I install the canary version the build step of chromatic-cli still uses npm.

**Comment by ghengeveld** [comment]
Released as `7.0.0-next.0`.

**Comment by ghengeveld** [comment]
@glancel Could you give `7.0.0-next.0` a try? If you use GitHub Actions, it's `chromaui/action-next`. I want to get some 👍 before I ship it as `7.0.0` final.

**Comment by glancel** [comment]
@ghengeveld It works great! I haven't tested the Github Action yet but the rest works great without any additional configuration!! Thanks a lot

**Comment by glancel** [comment]
@ghengeveld I'm trying to use the github action but I get the chromaui/action-next do not exist:
```yml 
name: 'Chromatic'
on:
  pull_request:
    types: [ opened, synchronize, reopened ]
    branches:
      - 'main'
env:
  NODE_VERSION: "18.x"
jobs:
  chromatic-deployment:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
        with:
          fetch-depth: 0
      - name: Setup Node ${{ env.NODE_VERSION }}
        uses: actions/setup-node@v3
        with:
          node-version: ${{ env.NODE_VERSION }}
      - name: Setup pnpm
        uses: pnpm/action-setup@v2
        with:
          version: latest
      - name: Install dependencies
        run: pnpm install
      - name: Publish to Chromatic
        uses: chromaui/action-next
        with:
          projectToken: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
```
Do you happen to know why this is happening?

**Comment by ghengeveld** [comment]
I think you needed `chromaui/action-next@v1` (with tag). I've already released it as "latest" though, so you can use `chromaui/action@v1` now.

**Comment by kevinlandsberg** [comment]
> LGTM. We'll follow up with PNPM + TurboSnap.

hey @thafryer , is there any status update about this topic? We are actually hitting a blocker by Chromatic not supporting TurboSnap in our current pnpm setup :(
We highly rely on Chromatic and love it, we really don't want to drop it but this way we cannot use it at all..
Or is there some kind of workaround?

**Pull request merged**